langkit/templates/pkg_implementation_body_ada.mako

```diff
@@ -1296,12 +1296,18 @@ package body ${ada_lib_name}.Implementation is
    function Text
      (Node : access ${root_node_value_type}'Class) return Text_Type is
    begin
+      --  No text is associated to synthetic and ghost nodes
+
       % if ctx.synthetic_nodes:
```

Comment:
Seeing this code, it makes me wonder if you'd not rather generate a `Synthetic_Nodes` subtype for which the predicate is just `False` when there are no synthetic nodes. Two reasons spring to mind:

1. First it would simplify this kind of code
2. Second it would go in the direction of being able to treat langkit generated libraries 'uniformly' from an user point of view. So you'd rather have a synthetic_nodes subtype that is an empty set, rather than not having it.

Reply:
Good idea, will do!